frontend/dockerfile/parser/errors.go

```diff
@@ -48,6 +65,56 @@ func WithLocation(err error, location []Range) error {
 	})
 }
 
+func MergeLocations(locations ...[]Range) []Range {
+	// normalize the range by looping through all the ranges
+	// and ensuring that the start and end are in the correct order
+	allRanges := []Range{}
+	for _, location := range locations {
+		for i, r := range location {
+			if r.Start.Line > r.End.Line {
```

Comment:
What's this case? `end > start` seems like invalid input.

Reply:
Yeah, it was meant to guard against invalid input (or rather, correct it). Probably is overzealous, so I'll trim it.